go.mod

```diff
@@ -1,6 +1,8 @@
 module kubevirt.io/containerized-data-importer
 
-go 1.22.0
+go 1.23.0
```

Comment:
Should we update go in the builder first?

Reply:
Probably, but it does appear to update the golang version in the builder at runtime now. Note that x/oauth version requires 1.23 to work.

Reply:
https://github.com/kubevirt/containerized-data-importer/pull/3650 updates the builder, once that is merged and a new builder available I will update this PR.